### PR TITLE
Working on `man.Rmd`

### PR DESCRIPTION
**Pull request by jennybc**
This comes from ~1 close reading and is just making changes that immediately come to mind.

(I might make a second PR with more changes (e.g. additions) or extend this one, if it's still open, once this marinates a bit more.)

### CODE REVIEW

File: man.Rmd [hunk 1]
```diff
@@ -161,51 +164,53 @@ The introduction provides a title, description, and, optionally, details, for th
...
--   `mutate()`: Create, modify, and delete columns.
--   `summarise()`: Summarize each group to fewer rows.
--   `filter()`: Subset rows using column values.
--   `select()`: Subset columns using their names and types.
--   `arrange()`: Arrange rows by column values.
+-   `mutate()`: Create, modify, and delete columns
+-   `summarise()`: Summarize each group to fewer rows
+-   `filter()`: Subset rows using column values
+-   `select()`: Subset columns using their names and types
+-   `arrange()`: Arrange rows by column values
```
Comment (jennybc) [inline man.Rmd:189]: I think it's worth leaving the trailing `.` off here, since that's what we want for the actual oxygen title.

File: man.Rmd [hunk 2]
```diff
@@ -161,51 +164,53 @@ The introduction provides a title, description, and, optionally, details, for th
...
 
 Here we've tried to succinctly describe what the function does, making sure to describe whether it affects rows, columns, or groups.
 We do our best to use synonyms, instead of repeating the function name, to hopefully give folks another chance to understand the intent of the function.
 
-At the time we wrote this chapter, I don't think the function titles for stringr were that successful.
+At the time we wrote this chapter, we found the function titles for stringr to be somewhat disappointing.
```
Comment (jennybc) [inline man.Rmd:194]: I generally got rid of "I" wording, with various techniques.

File: man.Rmd [hunk 3]
```diff
@@ -334,15 +340,15 @@ But instead of supplying both, you can use just `start` if you pass in a two-col
...
-#'   of the first character (defaults to first), `end` gives the position
-#'   of the last (defaults to last character). Alternatively, pass a two-column
-#'   matrix to `start`.
+#' @param start,end A pair of integer vectors defining the range of characters
+#'   to extract (inclusive).
 #'
-#'   Negative values count backwards from the last character.
+#'   Alternatively, instead a pair of vectors, you can pass a matrix to
+#'   `start`. The matrix should have two columns, either labelled `start`
+#'   and `end`, or `start` and `length`.
```
Comment (jennybc) [inline man.Rmd:348]: Porting updates that happened in stringr since this chapter was last touched.

File: man.Rmd [hunk 4]
```diff
@@ -451,23 +462,23 @@ For example `readr::read_csv()` mentions what happens if there are any parsing p
...
 
 -   Interactively using the `example()` function.
--   During `R CMD check` on your computer, or another computer you control (e.g. GitHub action).
+-   During `R CMD check` on your computer, or another computer you control (e.g. in GitHub Actions).
 -   During `R CMD check` run by CRAN.
--   When building your pkgdown website.
+-   When your pkgdown website is being built.
 
-After discussing what to put in your examples, we'll talk about keeping your examples self-contained, how to display errors if needed, handling dependencies, running examples conditionally, and
+After discussing what to put in your examples, we'll talk about keeping your examples self-contained, how to display errors if needed, handling dependencies, running examples conditionally, and WUT?!?
```
Comment (jennybc) [inline man.Rmd:481]: I'll come back to this.

File: man.Rmd [hunk 5]
```diff
@@ -505,33 +516,38 @@ Here's an example from `tidyr::chop()`:
 #' df <- tibble(x = 1:4, y = list(integer(), 1L, 1:2, 1:3))
 #' df %>% unchop(y)
 #' df %>% unchop(y, keep_empty = TRUE)
-#' 
-#' #' # Incompatible types -------------------------------------------------
+#'
+#' # Incompatible types ================================================
```
Comment (jennybc) [inline man.Rmd:520]: Jenny TODO: sync up with whatever comes of https://github.com/tidyverse/tidyr/pull/1412

File: man.Rmd [hunk 6]
```diff
@@ -505,33 +516,38 @@ Here's an example from `tidyr::chop()`:
...
-If you find yourself needing to do a bunch of setup to create a dataset or object to use in the example, it may be a sign that you need to create a package dataset.
-See @sec-data for details.
+If you can make the point with a familiar built-in dataset, like `mtcars`, do so.
+If you find yourself needing to do a bunch of setup to create a dataset or object to use in the example, it may be a sign that you need to create a package dataset or even a helper function.
+See @sec-data, @sec-data-example-path-helper, and @sec-testing-advanced-fixture-helper for ideas.
+Making it easy to write (and read) examples will greatly improve the quality of your documentation.
+
+Also, remember that examples are not tests.
+Examples should be focused on the authentic and typical usage you've designed for and that you want to encourage.
+The test suite is the more appropriate place to exhaustively exercise all of the arguments and to explore weird, pathological edge cases.
```
Comment (jennybc) [inline man.Rmd:534]: This is actual new content.

Comment (hadley) [reply man.Rmd:532]: It like it

**Review by hadley — APPROVED** [review]
Looks great!

File: man.Rmd [hunk 7]
```diff
@@ -2,21 +2,22 @@
...
-Instead, we'll use the roxygen2 package to generate them from specially formatted comments are written above each function.
+In the devtools ecosystem, we don't edit `.Rd` files directly with our bare hands.
+Instead, we include specially formatted comments *above* the source code for each function.
+Then we use the roxygen2 package to generate the `.Rd` files from these special "roxygen comments".
 There are a few advantages to using roxygen2:
 
--   Code and documentation are intermingled so that when you modify your code, it's easy to remember to also update your documentation.
+-   Code and documentation are co-located, so that when you modify your code, it's easy to remember to also update your documentation.
 
 -   You can use markdown, rather having to learn a new text formatting syntax.
```
Comment (jonthegeek) [inline man.Rmd:22]:
```suggestion
-   You can use markdown, rather than having to learn a new text formatting syntax.
```

**Review by jonthegeek — COMMENTED** [review]
I found a typo that slipped through.

I love the changes! I originally read this back before full RMarkdown support, so it's helpful to see more explicit information about what works now (rather than just picking things up via experimentation and reading code). I didn't realize Rmd sections work now!
